react/new/src/components/MainNavigation/main-navigation.module.scss

```diff
@@ -1,27 +1,216 @@
+  color: var(--mantine-color-text);
   font-size: var(--mantine-font-size-sm);
   font-weight: 500;
 }
 
+.menuLink {
+  text-decoration: none;
+  color: inherit;
+  display: block;
+  width: 100%;
+}
+
 .linkLabel {
-  margin-right: 0.3125rem; /* 5px / 16px */
+  margin-right: 0.3125rem;
+}
+
+.themeControls {
+  padding: 0.75rem;
+  min-width: 200px;
+}
+
+.mobileMenu {
+  display: none;
+  position: fixed;
+  top: 3.5rem;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: var(--mantine-color-body);
+  padding: 1rem;
+  z-index: 99;
+  overflow-y: auto;
+
+  @media (max-width: 768px) {
+    &.opened {
+      display: flex;
+      flex-direction: column;
+    }
+  }
+
+  .mobileLink {
+    color: var(--mantine-color-text);
+    text-decoration: none;
+    font-size: 1.25rem;
+    padding: 0.75rem;
+    border-radius: var(--mantine-radius-sm);
+    transition: background-color 150ms ease;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+
+    &:hover {
+      background-color: var(--mantine-color-dark-6);
+
+      [data-mantine-color-scheme="light"] & {
+        background-color: var(--mantine-color-gray-1);
+      }
+    }
+
+    &.hasChildren {
+      cursor: pointer;
+    }
+  }
+
+  .mobileSubmenu {
+    display: none;
+    padding-left: 1rem;
+    
+    &.opened {
+      display: flex;
+      flex-direction: column;
+      gap: 0.5rem;
+      padding-top: 0.5rem;
+    }
+
+    .mobileLink {
+      font-size: 1.1rem;
+      padding: 0.5rem 0.75rem;
+    }
+  }
+
+  .mobileDivider {
+    height: 1px;
+    background-color: var(--mantine-color-dark-4);
+    margin: 1rem 0;
+
+    [data-mantine-color-scheme="light"] & {
+      background-color: var(--mantine-color-gray-3);
+    }
+  }
+
+  .mobileSettings {
+    padding: 0.75rem;
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    color: var(--mantine-color-text);
+
+    .settingsHeader {
+      display: flex;
+      align-items: center;
+      gap: 0.75rem;
+      font-size: 1.25rem;
+      font-weight: 500;
+    }
+
+    .settingsContent {
+      padding-left: 0.25rem;
+      display: flex;
+      align-items: center;
+      gap: 0.75rem;
+      font-size: 1rem;
+    }
+  }
+}
+
+.rightGroup {
+  display: flex;
+  align-items: center;
+  gap: 2rem;
+  position: relative;
+}
+
+.settingsMenu {
+  @media (max-width: 768px) {
+    display: none;
+  }
+}
+
+.burgerWrapper {
+  @media (max-width: 768px) {
+    position: relative;
+    width: 40px;
+    height: 40px;
+    margin: -10px -10px -10px 0;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    background: none;
+    border: none;
+    padding: 0;
+    cursor: pointer;
+
+    .burger {
+      cursor: pointer;
+    }
+  }
+}
+
+:global(.mantine-Menu-dropdown) {
+  background-color: var(--mantine-color-body);
+  color: var(--mantine-color-text);
+  border: 1px solid var(--mantine-color-dark-4);
```

Comment:
I get the use of color-body and color-text for the theme service -- it's actually shockingly similar to my work team's theming. is color-dark-4 or color-gray-3 themed also? Or are those the same regardless of theme?